packages/next/src/server/request/search-params.ts

```diff
@@ -567,63 +568,37 @@ function makeDynamicallyTrackedExoticSearchParamsWithDevWarnings(
   })
 
   Object.keys(underlyingSearchParams).forEach((prop) => {
-    switch (prop) {
-      // Object prototype
-      case 'hasOwnProperty':
-      case 'isPrototypeOf':
-      case 'propertyIsEnumerable':
-      case 'toString':
-      case 'valueOf':
-      case 'toLocaleString':
-
-      // Promise prototype
-      // fallthrough
-      case 'then':
-      case 'catch':
-      case 'finally':
-
-      // React Promise extension
-      // fallthrough
-      case 'status':
-
-      // Common tested properties
-      // fallthrough
-      case 'toJSON':
-      case '$$typeof':
-      case '__esModule': {
-        // These properties cannot be shadowed because they need to be the
-        // true underlying value for Promises to work correctly at runtime
-        unproxiedProperties.push(prop)
-        break
-      }
-      default: {
-        proxiedProperties.add(prop)
-        Object.defineProperty(promise, prop, {
-          get() {
-            return proxiedUnderlying[prop]
-          },
-          set(newValue) {
-            Object.defineProperty(promise, prop, {
-              value: newValue,
-              writable: true,
-              enumerable: true,
-            })
-          },
-          enumerable: true,
-          configurable: true,
-        })
-      }
+    if (wellKnownProperties.has(prop)) {
+      // These properties cannot be shadowed because they need to be the
+      // true underlying value for Promises to work correctly at runtime
+      unproxiedProperties.push(prop)
+    } else {
+      proxiedProperties.add(prop)
+      Object.defineProperty(promise, prop, {
+        get() {
+          return proxiedUnderlying[prop]
+        },
+        set(newValue) {
+          Object.defineProperty(promise, prop, {
+            value: newValue,
+            writable: true,
+            enumerable: true,
+          })
+        },
+        enumerable: true,
+        configurable: true,
+      })
     }
   })
 
   const proxiedPromise = new Proxy(promise, {
     get(target, prop, receiver) {
       if (typeof prop === 'string') {
         if (
-          // We are accessing a property that was proxied to the promise instance
-          proxiedProperties.has(prop) ||
-          // We are accessing a property that doesn't exist on the promise nor the underlying
-          Reflect.has(target, prop) === false
+          !wellKnownProperties.has(prop) && // We are accessing a property that was proxied to the promise instance
+          (proxiedProperties.has(prop) ||
+            // We are accessing a property that doesn't exist on the promise nor the underlying
+            Reflect.has(target, prop) === false)
```

Comment:
@eps1lon 

Yeah I suppose this is the right approach. I was thinking maybe we could get rid of the warning if the property doesn't exist on either the promise or the underlying search params object but it is common to check for a search param to exist and if it doesn't you wouldn't get the warning in that case.

I think we can actually just simplify the logic to only warn if the prop is not part of the well known set regardless of whether it was proxied or not. The well known ones won't ever be in the proxied set so they are mutually exclusive and there 

so maybe

```suggestion
          // The property isn't part of the well known set that is never proxied
          !wellKnownProperties.has(prop) &&
          // We aren't accessing a property defined directly on the promise like 'status' or 'value' written by React
          Reflect.has(target, prop) === false
```

Reply:
This won't warn on the proxied properties though since we don't warn from the getter.